packages/gatsby/src/utils/page-ssr-module/bundle-webpack.ts

```diff
@@ -20,27 +19,15 @@ const extensions = [`.mjs`, `.js`, `.json`, `.node`, `.ts`, `.tsx`]
 const outputDir = path.join(process.cwd(), `.cache`, `page-ssr`)
 const cacheLocation = path.join(process.cwd(), `.cache`, `webpack`, `page-ssr`)
 
-export async function writeQueryContext({
-  staticQueriesByTemplate,
-  components,
-}: {
+// eslint-disable-next-line
+export async function copyStaticQueriesToEngine(_args: {
+  engineTemplatePaths: Set<string>
   staticQueriesByTemplate: IGatsbyState["staticQueriesByTemplate"]
-  components: IGatsbyState["components"]
 }): Promise<void> {
-  const waitingForWrites: Array<Promise<unknown>> = []
-  for (const pageTemplate of components.values()) {
-    const staticQueryHashes =
-      staticQueriesByTemplate.get(pageTemplate.componentPath) || []
-
-    waitingForWrites.push(
-      writeStaticQueryContext(
-        staticQueryHashes,
-        pageTemplate.componentChunkName
-      )
-    )
-  }
-
-  return Promise.all(waitingForWrites).then(() => {})
+  // TODO: collect static queries for engine template paths (need to traverse slices used by a template too)
+  const sourceDir = path.join(process.cwd(), `public`, `page-data`, `sq`, `d`)
+  const destDir = path.join(outputDir, `sq`)
+  await fs.copy(sourceDir, destDir)
```

Comment:
this fails `gatsby-cli` integration tests as fixture sites seem to not create some directories - this will be fixed as part of the TODO